vpn/muxer.go

```diff
@@ -403,9 +449,15 @@ func (m *muxer) readPushReply() error {
 		return fmt.Errorf("%w:%s", errBadServerReply, "expected push reply")
 	}
 
-	ip := m.control.ReadPushResponse(resp)
-	m.tunnel.ip = ip
-	logger.Infof("Tunnel IP: %s", ip)
+	optsMap := m.control.ReadPushResponse(resp)
+	ti := newTunnelInfoFromPushedOptions(optsMap)
```

Comment:
could parsing fail? do we care about that?

Reply:
same answer as in the function above. #30 adds a few test cases for this function, so deferring closer examination to that one.